02-kotlin-helloworld/src/test/kotlin/HelloProtelisTests.kt

```diff
@@ -0,0 +1,70 @@
+
+package demo
+
+import io.kotlintest.Spec
+import io.kotlintest.TestCase
+import io.kotlintest.matchers.exactly
+import io.kotlintest.specs.FunSpec
+import io.kotlintest.specs.StringSpec
+import io.mockk.confirmVerified
+import io.mockk.spyk
+import io.mockk.verify
+import org.jgrapht.Graphs
+import org.jgrapht.graph.DefaultEdge
+import org.jgrapht.graph.DefaultUndirectedGraph
+import org.protelis.lang.ProtelisLoader
+
+class HelloProtelisTests : StringSpec() {
+
+    val protelisModuleName = "hello"
+    val n = 3
+    val iterations = 4
+    var devices: List<Device> = emptyList()
+    var speakers: List<Speaker> = emptyList()
+    val leader = 0
+
+    override fun beforeSpec(spec: Spec) {
+        val g = DefaultUndirectedGraph<Device, DefaultEdge>(DefaultEdge::class.java)
+        repeat(n) {
+            val program = ProtelisLoader.parse(protelisModuleName)
+            val s = spyk(ConsoleSpeaker())
+            val d = Device(program, it, EmulatedNetworkManager(IntDeviceUID(it)), s)
+            devices = devices + d
+            speakers = speakers + s
+            g.addVertex(d)
+        }
+        repeat(n) {
+            g.addEdge(
+                    devices[it],
+                    devices[(it + 1) % n])
+        }
+
```

Comment:
Do not leave uncommented empty lines. Either provide a comment to justify the separation or keep a contiguous flow of instructions